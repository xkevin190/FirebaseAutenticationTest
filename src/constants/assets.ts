export const FONTS = {
    LATO: 'Lato',
};