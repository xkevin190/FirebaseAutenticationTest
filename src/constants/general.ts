export const SCREEN_ANIMATION = 'slide_from_right';

export const APIS = {
  USERS: {
    GET_IN_TOUCH: 'users/getInTouch',
  },
  AUTH: {
    CREATE_ACCOUNT: 'auth/parents/signup',
    LOGIN: 'auth/parents/login',
    RECOVER_PASSWORD: 'auth/recover',
  },
};

export const NO_CONNECTION_ERROR = 'TypeError: Network request failed';
