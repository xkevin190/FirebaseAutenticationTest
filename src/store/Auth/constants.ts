export type IAccountActionTypes = {
  CREATE_ACCOUNT: string;
  LOGIN: string;
  ClOSE_SESSION: string;
};

export const AccounthActionsTypes: IAccountActionTypes = {
  CREATE_ACCOUNT: 'CREATE_ACCOUNT',
  LOGIN: ' LOGIN',
  ClOSE_SESSION: 'ClOSE_SESSION',
};
