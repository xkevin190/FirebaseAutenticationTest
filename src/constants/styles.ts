import { FONTS } from "./assets";

export const COLORS = {
    WHITE: '#FFFFFF',
    BLACK: '#000000',
    BLACK_025: '#00000040',
    GRAY_DEFAULT: '#46484B',
    GRAY_DEFAULT_DARKER: '#646565',
    GRAY_MEDIUM: '#9DA5B9',
    GRAY_LIGHT: '#F6F8FF',
    GRAY_LIGHT_DARKER: '#DDDDDD',
    DISABLED_DEFAULT: '#DFE6ED',
    OPYA_BLUE: '#2162E3',
    OPYA_VIOLET: '#1745D5',
    MAIN_DEFAULT: '#313342',
    SECONDARY_LIGHT: '#62698B',
    ALERT_DEFAULT: '#D15686',
    ALERT_LIGHT: '#FFD0E2',
  };

  const FONTS_TYPES = {
    BLACK: 'Black',
    BLACK_ITALIC: 'BlackItalic',
    BOLD: 'Bold',
    BOLD_ITALIC: 'BoldItalic',
    HAIR_LINE: 'HairLine',
    HAIR_LINE_ITALIC: 'HairLineItalic',
    HEAVY: 'Heavy',
    HEAVY_ITALIC: 'HeavyItalic',
    ITALIC: 'Italic',
    LIGHT: 'Light',
    LIGHT_ITALIC: 'LightItalic',
    MEDIUM: 'Medium',
    MEDIUM_ITALIC: 'MediumItalic',
    REGULAR: 'Regular',
    SEMI_BOLD: 'SemiBold',
    SEMI_BOLD_ITALIC: 'SemiBoldItalic',
    THIN: 'Thin',
    THIN_ITALIC: 'ThinItalic',
  };


  export const STYLES = {
    FONT_FAMILY: {
      LATO_REGULAR: `${FONTS.LATO}-${FONTS_TYPES.REGULAR}`,
      LATO_BOLD: `${FONTS.LATO}-${FONTS_TYPES.BOLD}`,
    },
    FONT_SIZE: {
      PARAGRAPH: 14,
      SMALL_PARAGRAPH: 12,
      MEDIUM_PARAGRAPH: 16,
      INTERMEDIUM_PARAGRAPH: 18,
      LARGE_PARAGRAPH: 20,
    },
  };
  

