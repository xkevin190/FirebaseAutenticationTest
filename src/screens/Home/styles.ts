import {StyleSheet, ViewStyle} from 'react-native';

import {COLORS} from '../../constants/styles';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    backgroundColor: COLORS.GRAY_LIGHT,
  } as ViewStyle,
});

export default styles;
