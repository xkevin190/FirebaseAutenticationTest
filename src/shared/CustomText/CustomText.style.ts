import {StyleSheet, TextStyle} from 'react-native';

import {STYLES} from '../../constants/styles';

const styles = StyleSheet.create({
  'small-paragraph-bold': {
    fontFamily: STYLES.FONT_FAMILY.LATO_BOLD,
    fontSize: STYLES.FONT_SIZE.SMALL_PARAGRAPH,
  } as TextStyle,
  'small-paragraph': {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.SMALL_PARAGRAPH,
  } as TextStyle,
  'medium-paragraph': {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.MEDIUM_PARAGRAPH,
  } as TextStyle,
  paragraph: {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.PARAGRAPH,
  } as TextStyle,
  'large-paragraph': {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.LARGE_PARAGRAPH,
  } as TextStyle,
  'paragraph-bold': {
    fontFamily: STYLES.FONT_FAMILY.LATO_BOLD,
    fontSize: STYLES.FONT_SIZE.PARAGRAPH,
  } as TextStyle,
  'intermedium-paragraph-bold': {
    fontFamily: STYLES.FONT_FAMILY.LATO_BOLD,
    fontSize: STYLES.FONT_SIZE.INTERMEDIUM_PARAGRAPH,
  } as TextStyle,
  'intermedium-paragraph': {
    fontSize: STYLES.FONT_SIZE.INTERMEDIUM_PARAGRAPH,
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
  } as TextStyle,
});

export default styles;
