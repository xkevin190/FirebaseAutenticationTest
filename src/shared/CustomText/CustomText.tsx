import React from 'react';
import {Text, TextStyle} from 'react-native';

import {COLORS} from '../../constants/styles';
import {TextBody} from '../../types/CustomText';
import styles from './CustomText.style';

export interface ICustomTextProps {
  text: string;
  center?: boolean;
  color?: string;
  bodyType?: TextBody;
  testID?: string;
  customStyle?: TextStyle;
}

const CustomText: React.FC<ICustomTextProps> = ({
  text,
  color = COLORS.BLACK,
  bodyType = 'paragraph',
  center = false,
  testID = '',
  customStyle = {},
}) => (
  <Text
    style={{
      ...styles[bodyType],
      ...customStyle,
      ...(center && {textAlign: 'center'}),
      color,
    }}
    testID={testID}>
    {text}
  </Text>
);

export default CustomText;
