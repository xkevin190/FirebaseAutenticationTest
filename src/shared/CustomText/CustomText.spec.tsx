import React from 'react';
import {TextProps, TextStyle} from 'react-native';
import {shallow, ShallowWrapper} from 'enzyme';

import styles from './CustomText.style';
import {COLORS} from '../../constants/styles';
import CustomText, {ICustomTextProps} from './CustomText';

describe('CustomText (Unit-Test)', () => {
  const props: ICustomTextProps = {
    text: 'CUSTOM_TEXT',
  };
  let wrapper: ShallowWrapper;

  describe('general functionality', () => {
    beforeEach(() => {
      wrapper = shallow(<CustomText {...props} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render "text" prop', () => {
      const textProps = wrapper.find('Text').props() as TextProps;
      expect(textProps.children).toEqual(props.text);
    });

    it('should render default button style', () => {
      const textProps = wrapper.find('Text').props() as TextProps;
      expect(textProps.style).toEqual({
        ...styles.paragraph,
        color: COLORS.BLACK,
      });
    });

    it('should set "testID" with empty string', () => {
      const textProps = wrapper.find('Text').props() as TextProps;
      expect(textProps.testID).toEqual('');
    });
  });

  describe('when optional props are received', () => {
    it('should render text color with "color" prop received', () => {
      wrapper = shallow(
        <CustomText {...props} color={COLORS.GRAY_LIGHT_DARKER} />,
      );
      const textProps = wrapper.find('Text').props() as TextProps;
      expect(textProps.style).toEqual({
        ...styles.paragraph,
        color: COLORS.GRAY_LIGHT_DARKER,
      });
    });

    it('should render text aligned center when "center" prop received', () => {
      wrapper = shallow(<CustomText {...props} center />);
      const textProps = wrapper.find('Text').props() as TextProps;
      expect(textProps.style).toEqual({
        ...styles.paragraph,
        color: COLORS.BLACK,
        textAlign: 'center',
      });
    });

    it('should render text with styles depending of "bodyType" prop received', () => {
      wrapper = shallow(<CustomText {...props} bodyType="small-paragraph" />);
      const textProps = wrapper.find('Text').props() as TextProps;
      expect(textProps.style).toEqual({
        ...styles['small-paragraph'],
        color: COLORS.BLACK,
      });
    });

    it('should set testID with "testID" prop received', () => {
      wrapper = shallow(<CustomText {...props} testID="TESTING_TEST_ID" />);
      const textProps = wrapper.find('Text').props() as TextProps;
      expect(textProps.testID).toEqual('TESTING_TEST_ID');
    });

    it('should add new styles when "customStyles" prop is received', () => {
      const customStyles: TextStyle = {
        lineHeight: 10,
        margin: 20,
      };
      wrapper = shallow(
        <CustomText
          {...props}
          bodyType="small-paragraph"
          customStyle={customStyles}
        />,
      );
      const textProps = wrapper.find('Text').props() as TextProps;
      expect(textProps.style).toEqual({
        ...styles['small-paragraph'],
        ...customStyles,
        color: COLORS.BLACK,
      });
    });
  });
});
