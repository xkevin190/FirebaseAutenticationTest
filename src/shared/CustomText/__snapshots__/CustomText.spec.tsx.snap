// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText (Unit-Test) general functionality should match snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#000000",
      "fontFamily": "Lato-Regular",
      "fontSize": 14,
    }
  }
  testID=""
>
  CUSTOM_TEXT
</Text>
`;
